stock_quantity_history_location/static/src/components/inventory_report/inventory_report.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template" xml:space="preserve">
+    <t t-inherit="stock.InventoryReport.Buttons" t-inherit-mode="extension" owl="1">
+        <xpath expr="(//div/*)[last()]" position="after">
```

Comment:
AFAIK this is inserting elements at the end, even after the "Appli All" button.

Maybe what you need is to place this before the "Apply All" button, by filtering by `t-on-click="onClickApplyAll"`.